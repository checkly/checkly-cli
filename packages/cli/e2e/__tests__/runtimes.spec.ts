import { runChecklyCli } from '../run-checkly'

describe('runtimes', () => {
  it('should return supported runtimes', async () => {
    const { stdout } = await runChecklyCli({
      args: ['runtimes'],
    })
    expect(stdout).toMatchSnapshot('runtimes')
  })
})
