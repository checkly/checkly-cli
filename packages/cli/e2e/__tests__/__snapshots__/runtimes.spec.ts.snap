// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runtimes should return supported runtimes: runtimes 1`] = `
"
Notice: replacing version '0.0.1-dev' with latest '4.0.14'.

[
  {
    "name": "2023.02",
    "description": "Main updates are Playwright 1.32.1, faker 7.6.0 and the addition of date-fns 2.29.3 and ws 8.13.0. We are also dropping support for Mocha.",
    "stage": "CURRENT",
    "nodeVersion": "16.x",
    "dependencies": {
      "@faker-js/faker": "7.6.0",
      "@google-cloud/local-auth": "2.1.1",
      "@opentelemetry/api": "1.0.4",
      "@opentelemetry/sdk-trace-base": "1.0.1",
      "@playwright/test": "1.32.3",
      "aws4": "1.11.0",
      "axios": "0.27.2",
      "btoa": "1.2.1",
      "chai": "4.3.7",
      "chai-string": "1.5.0",
      "crypto-js": "4.1.1",
      "date-fns": "2.29.3",
      "date-fns-tz": "2.0.0",
      "expect": "29.3.1",
      "form-data": "4.0.0",
      "gmail-api-parse-message-ts": "2.2.32",
      "google-auth-library": "8.8.0",
      "googleapis": "118.0.0",
      "jose": "4.14.1",
      "jsdom": "21.1.2",
      "jsonwebtoken": "9.0.0",
      "lodash": "4.17.21",
      "moment": "2.29.2",
      "otpauth": "9.0.2",
      "twilio": "4.11.1",
      "playwright": "1.32.3",
      "typescript": "4.8.4",
      "uuid": "9.0.0",
      "ws": "8.13.0",
      "xml-crypto": "4.0.1",
      "xml-encryption": "3.0.2"
    }
  },
  {
    "name": "2022.10",
    "description": "Main updates are Playwright 1.28.0, Node.js 16.x and Typescript support. We are also dropping support for Puppeteer",
    "stage": "STABLE",
    "nodeVersion": "16.x",
    "dependencies": {
      "@faker-js/faker": "5.5.3",
      "@opentelemetry/api": "1.0.4",
      "@opentelemetry/sdk-trace-base": "1.0.1",
      "@playwright/test": "1.28.0",
      "aws4": "1.11.0",
      "axios": "0.27.2",
      "btoa": "1.2.1",
      "chai": "4.3.7",
      "chai-string": "1.5.0",
      "crypto-js": "4.1.1",
      "expect": "29.3.1",
      "form-data": "4.0.0",
      "jsonwebtoken": "8.5.1",
      "lodash": "4.17.21",
      "mocha": "10.1.0",
      "moment": "2.29.2",
      "otpauth": "9.0.2",
      "playwright": "1.28.0",
      "typescript": "4.8.4",
      "uuid": "9.0.0"
    }
  },
  {
    "name": "2022.02",
    "description": "Main updates are Playwright 1.20.2 and Puppeteer 13.7.0 and Node.js 14.x",
    "stage": "STABLE",
    "nodeVersion": "14.x",
    "dependencies": {
      "@faker-js/faker": "5.5.3",
      "@opentelemetry/api": "1.0.4",
      "@opentelemetry/sdk-trace-base": "1.0.1",
      "@playwright/test": "1.20.2",
      "aws4": "1.11.0",
      "axios": "0.26.0",
      "btoa": "1.2.1",
      "chai": "4.3.4",
      "chai-string": "1.5.0",
      "crypto-js": "4.1.1",
      "expect": "27.5.1",
      "form-data": "4.0.0",
      "jsonwebtoken": "8.5.1",
      "lodash": "4.17.21",
      "mocha": "9.2.1",
      "moment": "2.29.2",
      "playwright": "1.20.2",
      "puppeteer": "13.7.0",
      "request": "2.88.2",
      "request-promise": "4.2.6",
      "totp-generator": "0.0.13",
      "uuid": "8.3.2"
    }
  },
  {
    "name": "2021.10",
    "description": "Main updates are Playwright 1.15.2 and Puppeteer 10.4.0 and Node.js 14.x",
    "stage": "STABLE",
    "nodeVersion": "14.x",
    "dependencies": {
      "aws4": "1.11.0",
      "axios": "0.23.0",
      "btoa": "1.2.1",
      "chai": "4.3.4",
      "chai-string": "1.5.0",
      "crypto-js": "4.1.1",
      "expect": "27.3.1",
      "form-data": "4.0.0",
      "faker": "5.5.3",
      "jsonwebtoken": "8.5.1",
      "lodash": "4.17.21",
      "mocha": "9.1.3",
      "moment": "2.29.2",
      "playwright": "1.15.2",
      "puppeteer": "10.4.0",
      "request": "2.88.2",
      "request-promise": "4.2.2",
      "uuid": "8.3.2"
    }
  }
]
"
`;
